prime-router/src/test/kotlin/azure/SettingsFunctionTests.kt

```diff
@@ -0,0 +1,581 @@
+        httpRequestMessage.httpHeaders += mapOf(
+            "authorization" to "Bearer 111.222.333",
+            "content-type" to HttpUtilities.jsonMediaType
+        )
+
+        return httpRequestMessage
+    }
+
+    @Test
+    fun `test access user cannot view full organization settings (getOrganizations)`() {
+        val settingsFunction = setupSettingsFunctionForTesting(oktaClaimsOrganizationName, mockFacade())
+        val httpRequestMessage = setupHttpRequestMessageForTesting()
+        val response = settingsFunction.getOrganizations(httpRequestMessage)
+        assertThat(response.status).isEqualTo(HttpStatus.UNAUTHORIZED)
+    }
+
+    @Test
+    fun `test access DHPrimeAdmins can view all organization settings (getOrganizations)`() {
+        val settingsFunction = setupSettingsFunctionForTesting("DHPrimeAdmins", mockFacade())
+        val httpRequestMessage = setupHttpRequestMessageForTesting()
+        var response = settingsFunction.getOrganizations(httpRequestMessage)
+        assertThat(response.status).isEqualTo(HttpStatus.OK)
+        response = settingsFunction.getOrganizations(httpRequestMessage)
+        assertThat(response.status).isEqualTo(HttpStatus.OK)
+    }
+
+    @Test
+    fun `test access user cannot view their organization's settings (getOneOrganization)`() {
+        val settingsFunction = setupSettingsFunctionForTesting(oktaClaimsOrganizationName, mockFacade())
+        val httpRequestMessage = setupHttpRequestMessageForTesting()
+        val response = settingsFunction.getOneOrganization(
+            httpRequestMessage,
+            "$organizationName.elr-secondary"
+        )
+        assertThat(response.status).isEqualTo(HttpStatus.UNAUTHORIZED)
+    }
+
+    @Test
+    fun `test access user cannot view another organization's settings (getOneOrganization)`() {
+        val settingsFunction = setupSettingsFunctionForTesting(oktaClaimsOrganizationName, mockFacade())
+        val httpRequestMessage = setupHttpRequestMessageForTesting()
+        val response = settingsFunction.getOneOrganization(
+            httpRequestMessage,
+            "$otherOrganizationName.test-lab-2"
+        )
+        assertThat(response.status).isEqualTo(HttpStatus.UNAUTHORIZED)
+    }
+
+    @Test
+    fun `test access DHPrimeAdmins can view any organization's settings (getOneOrganization)`() {
+        val settingsFunction = setupSettingsFunctionForTesting("DHPrimeAdmins", mockFacade())
+        val httpRequestMessage = setupHttpRequestMessageForTesting()
+        var response = settingsFunction.getOneOrganization(
+            httpRequestMessage,
+            "$organizationName.elr-secondary"
+        )
+        assertThat(response.status).isEqualTo(HttpStatus.OK)
+        response = settingsFunction.getOneOrganization(
+            httpRequestMessage,
+            "$otherOrganizationName.test-lab-2"
+        )
+        assertThat(response.status).isEqualTo(HttpStatus.OK)
```

Comment:
So this assertion is sort of correct, as the admin is authorized, but because you have a mock that always returns `""` for any organization name (89L) the body returned in the response will be `""`.  In production this would have returned a `404`

Reply:
Right... but the assertion is incorrect here? `assertThat(response.status).isEqualTo(HttpStatus.OK)` would actually be a 404?